interview-questions.md

```diff
@@ -7,44 +7,69 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 3. Name three rails generator commands. What is created by each?
 
   Your answer:
+  rails generate model: will create a model in the database
+  rails generate migration: will allow modifications to a database
+  rails generate rspec:install: will add dependencies for RSPEC testing
 
   Researched answer:
-
+  rails generate model: will create a model in the database. When generating a model we can specify column names and datatypes that will go into the database. In the application/models folder a new file will be created for the model. It also creates a migration file in the db folder. The table information can be seen in the schema of the application. 
+  rails generate migration: allows you to modify the information in the database. This will be reflected in the schema of the application.
+  rails generate rspec:install: will add dependencies for RSPEC testing. In order to add dependencies for RSPEC testing we would need to run "bundle add rspec-rails" before running "rails generate rspec:install". Together these commands will add dependencies for RSPEC testing in Active Record models.
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
 action: "GET"    location: /students          
+This Rails route would call the index controller method and it will display a list of all the students.
 
 action: "POST"   location: /students       
+This Rails route would call on the create method and it will create a new instance of student by adding the data to the database.
 
 action: "GET"    location: /students/new
+This Rails route would call on the new method and it will display a form to a user where they can input data. This method is used in conjunction with the create method to create a new instance of student.
 
 action: "GET"    location: /students/2  
+This Rails route would call on the show method and it will display the student information with the id of 2.
 
-action: "GET"    location: /students/2/edit    
+action: "GET"    location: /students/2/edit  
+This Rails route would call on the edit method and it will display a form to a user where they can edit information about the student with id 2. This method is used in conjunction with the update method.  
 
-action: "PATCH"  location: /students/2      
+action: "PATCH"  location: /students/2   
+This Rails route would call on the update method and it will take the information edited about the student with id 2 in the form and update the database.  
 
 action: "DELETE" location: /students/2      
-
+This Rails route would call on the destroy method and it will delete the student entry with id 2.
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```

Comment:
User stories are tricky especially after we have learned how to develop. When writing user stories, think of what the user sees. For instance, As a User, I can delete a task from my list.